### PR TITLE
Add disable escaping option for json parsing. Fix string escaping for event-logs.

### PR DESCRIPTION
**Pull request by pujagani**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Add disable escaping option for json parsing. 

### Motivation and Context

The event-logs attributes like session payload and capabilities leverage the built-in json parse to convert the objects to json strings. By default, the escape characters are included in the string. It resulted in inconsistent log string with some fields having escape characters and some being printed normally. The changes added, fix this by adding a disable escaping option. 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: java/client/src/org/openqa/selenium/json/JsonOutput.java [hunk 1]
```diff
@@ -65,38 +65,6 @@
     GSON_ELEMENT = gsonElement;
   }
 
-  // https://www.json.org has some helpful comments on characters to escape
-  // See also https://tools.ietf.org/html/rfc8259#section-7 and
-  // https://github.com/google/gson/issues/341 so we escape those as well.
-  // It's legal to escape any character, so to be nice to HTML parsers,
-  // we'll also escape "<" and "&"
-  private static final Map<Integer, String> ESCAPES;
```
Comment (shs96c) [inline java/client/src/org/openqa/selenium/json/JsonOutput.java:73]: Prefer keeping this static block and setting a boolean for escaping if we're heading down this route.

File: java/client/src/org/openqa/selenium/json/JsonOutput.java [hunk 2]
```diff
@@ -213,6 +184,42 @@ public JsonOutput setPrettyPrint(boolean enablePrettyPrinting) {
     return this;
   }
 
+  // https://www.json.org has some helpful comments on characters to escape
+  // See also https://tools.ietf.org/html/rfc8259#section-7 and
+  // https://github.com/google/gson/issues/341 so we escape those as well.
+  // It's legal to escape any character, so to be nice to HTML parsers,
+  // we'll also escape "<" and "&"
+  private void setEscapes() {
```
Comment (shs96c) [inline java/client/src/org/openqa/selenium/json/JsonOutput.java:192]: Doing this per-json encoder is wasteful.

**Review by shs96c — CHANGES_REQUESTED** [review]
I'm not sure making the JSON encoder non-conformant is the right approach to follow for this problem. AIUI, we're passing in a json-encoded string to a logger, and then that's being escaped again. Perhaps we should investigate why we're doing the json escaping instead?

**Comment by barancev** [comment]
Can you please demonstrate what logging issue we're trying to solve by this change?

**Comment by pujagani** [comment]
Thank you for the review. Closing this PR in lieu of investigating the escaped string sent to the logger for event-logs.

**Comment by pujagani** [comment]
The logs currently are printed as follows: 
`{"traceId": "fc2964f4b3934af0a48df04e812acf5b","spanId": "dc660fec12498442","spanKind": "INTERNAL","eventTime": 1611575427636017843,"eventName": "Session request received by the distributor","attributes": {"logger": "org.openqa.selenium.grid.distributor.local.LocalDistributor","request.payload": "{\n  \"desiredCapabilities\": {\n    \"browserName\": \"chrome\"\n  },\n  \"capabilities\": {\n    \"firstMatch\": [\n      {\n        \"browserName\": \"chrome\"\n      }\n    ]\n  }\n}"}}` 

The reason is we leverage OpenTelemetry events and event attributes to display event-logs. Since, OpenTelemetry only accepts a string for an attribute value, for objects like payload and capabilities, we convert them to json string and add as an even-attribute. When the log statement is created, the fields from the span and events are added to a map and converted to a json string. Thus, object strings are escaped twice. The PR was aiming to as disable escaping option for such cases.

**Pull request closed without merging**